arrow-core-data/src/main/kotlin/arrow/core/Validated.kt

```diff
@@ -217,29 +250,28 @@ typealias Invalid<E> = Validated.Invalid<E>
  * }
  *
  * data class Config(val map: Map<String, String>) {
- *  fun <A> parse(read: Read<A>, key: String): Validated<ConfigError, A> {
- *   val v = Option.fromNullable(map[key])
- *   return when (v) {
- *    is Some ->
- *     when (val s = read.read(v.t)) {
- *      is Some -> s.t.valid()
- *      is None -> ConfigError.ParseConfig(key).invalid()
- *     }
- *    is None -> Validated.Invalid(ConfigError.MissingConfig(key))
+ *   suspend fun <A> parse(read: Read<A>, key: String) = Validated.fx<ConfigError, A> {
+ *     val value = Validated.fromNullable(map[key]) {
+ *       ConfigError.MissingConfig(key)
+ *     }.bind()
+ *     val readVal = Validated.fromOption(read.read(value)) {
+ *       ConfigError.ParseConfig(key)
+ *     }.bind()
+ *     readVal
  *   }
- *  }
  * }
  *
- * fun <E, A, B, C> parallelValidate(v1: Validated<E, A>, v2: Validated<E, B>, f: (A, B) -> C): Validated<E, C> {
- *  return when {
+ * fun <E, A, B, C> parallelValidate
+ *   (v1: Validated<E, A>, v2: Validated<E, B>, f: (A, B) -> C): Validated<NonEmptyList<E>, C> =
+ *  when {
  *   v1 is Validated.Valid && v2 is Validated.Valid -> Validated.Valid(f(v1.a, v2.a))
- *   v1 is Validated.Valid && v2 is Validated.Invalid -> v2
- *   v1 is Validated.Invalid && v2 is Validated.Valid -> v1
- *   v1 is Validated.Invalid && v2 is Validated.Invalid -> TODO()
- *   else -> TODO()
+ *   v1 is Validated.Valid && v2 is Validated.Invalid -> v2.toValidatedNel()
+ *   v1 is Validated.Invalid && v2 is Validated.Valid -> v1.toValidatedNel()
+ *   v1 is Validated.Invalid && v2 is Validated.Invalid -> Validated.Invalid(NonEmptyList(v1.e, listOf(v2.e)))
+ *   else -> throw IllegalStateException("Not possible value")
```

Comment:
Instead of `Throw`, we could return an Invalid with it.

Reply:
But that's the impossible case, does it really matter? Actually I was looking for if there was an operator that does this already and doesn't leave the burden of writing this to the dev but couldn't see any

Reply:
Isn't it exhaustive then?

Reply:
It cannot be because we're checking 2 elements

Reply:
we need pattern matching *weep*

Reply:
Cannot we add a mapN-like function to do this already? This way we could hide the throwing under our own impl cc @nomisRev

Reply:
Yes, you can use `Applicative` for this. 

```
Validated.applicative(NonEmptyList.semigroup<String>()).tupledN(
  1.valid(),
  "2".invalidNel(),
  "3".invalidNel()
) // Validated.Invalid(NonEmptyList("2", "3"))
```

Reply:
Right, I just realized that we're in core-data and we don't have access to the extensions :/

What I did is the following:

* Added a convenient `nelApplicative` function in Validated extensions
* Suggest to use this approach instead of the custom function in docs, but without changing the examples code due to the import problem

Reply:
It's just meh in general. It could be possible to rewrite it as two nested folds and some manual formatting.

Reply:
I'm not a fan of showing `parallelValidate` at all.. It promotes/shows code that no-one should ever write using Arrow.

We have a bunch of `helpers` in the `docs` module, any way we can provide instances there to make the desired snippets compile?

Reply:
The only reason I like it is for the explanation vs a Validated Nel, which is also explained in there. I changed it to try to make it clear that this is just for the shake of the example